show_graph.py

```diff
@@ -0,0 +1,148 @@
+  envoy_means = list()
+  envoy_std = list()
+
+  for single_set in direct_lists:
+    direct_means.append(np.mean(single_set))
+    direct_std.append(np.std(single_set))
+  for single_set in envoy_lists:
+    envoy_means.append(np.mean(single_set))
+    envoy_std.append(np.std(single_set))
+
+  width = 0.35
+  fig, ax = plt.subplots()
+  rects1 = ax.bar(ind, direct_means, width, color="r", yerr=direct_std)
+  rects2 = ax.bar(ind + width, envoy_means, width, color="y", yerr=envoy_std)
+
+  ax.set_ylabel(y_axis_field)
+  ax.set_xlabel(x_axis_field)
+  ax.set_xticks(ind + width)
+  ax.set_xticklabels(x_axis_values)
+  ax.legend((rects1[0], rects2[0]), ("Direct", "Envoy"))
+  AutoLabel(rects1, ax)
+  AutoLabel(rects2, ax)
+  fig.savefig("{} {}.pdf".format(
+      x_axis_field, ",".join(str(i) for i in x_axis_values)),
+              bbox_inches="tight")
+
+
+def AutoLabel(rects, ax):
+  """Attach a text label above each bar displaying its height.
+
+  This function is taken from matplotlib Demo.
+  Args:
+    rects: the rectangle
+    ax: the axis
+  """
+  for rect in rects:
+    height = rect.get_height()
+    ax.text(rect.get_x() + rect.get_width()/2., 1.05*height,
+            "%d " % int(height),
+            ha="center", va="bottom")
+
+
+def main():
+  parser = argparse.ArgumentParser(
+      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  parser.add_argument("--db_instance_name",
+                      help="the name of the gcloud instance",
+                      default="envoy-db-instance")
+  parser.add_argument("--username", help="username on the host DB",
+                      default="root")
+  parser.add_argument("--database", help="name of the database",
+                      default="envoy_stat_db")
+  parser.add_argument("--table_name", help=("the table which stores "
+                                            "the benchmarking data"),
+                      default="envoy_stat")
+  parser.add_argument("--project",
+                      help="the project name in Google Cloud.",
+                      default="envoy-ci")
+  parser.add_argument("--logfile",
+                      help="the local log file for this script. New log will be"
+                           "appended to this file.", default="benchmark.log")
+  parser.add_argument("--envoy_hashes",
+                      help="the hash of envoy version",
+                      nargs="+")
+  parser.add_argument("--runids",
+                      help=("the run ids to be plotted on the graph."
+                            " Give in a space-separated format."),
+                      nargs="+")
+  parser.add_argument("--fieldname",
+                      help="the name of the field for which you want the graph",
+                      default="total_req_per_sec")
+
+  args = parser.parse_args()
+  logfile = open(args.logfile, "ab")
+  hostname = db_utils.GetInstanceIP(args.db_instance_name, args.project)
+
+  password = utils.GetRandomPassword()
```

Comment:
This is actually authorization to the database. As this graph generation script can be run from any other machine, not only the benchmarking machine, I needed to authorize the DB again for the new machine. Instead of saving the previous random password from the previous scripts, I thought of setting a new random password as safer.